sample.wdl

```diff
@@ -19,42 +24,104 @@ workflow Sample {
         File? refflatFile
         Boolean variantCalling = false
         Map[String, String] dockerImages
+        String platform = "illumina"
+        Array[File]+? targetIntervals
+        File? ampliconIntervals
     }
 
-    scatter (lib in sample.libraries) {
-        call libraryWorkflow.Library as library {
+    scatter (readgroup in sample.readgroups) {
+        String libDir = outputDir + "/lib_" + readgroup.lib_id
+        String readgroupDir = libDir + "/rg_" + readgroup.id
+        String rgLine ='"ID:${readgroup.id}" "LB:${readgroup.lib_id}" "PL:${platform}" "SM:${sample.id}"'
+        call qcWorkflow.QC as qc {
             input:
-                reference = reference,
-                dbsnp = dbsnp,
-                starIndex = starIndex,
-                hisat2Index = hisat2Index,
-                strandedness = strandedness,
-                refflatFile = refflatFile,
-                outputDir = outputDir + "/lib_" + lib.id,
-                sample = sample,
-                library = lib,
-                variantCalling = variantCalling,
+                outputDir = readgroupDir,
+                read1 = readgroup.R1,
+                read2 = readgroup.R2,
                 dockerImages = dockerImages
         }
-        File lbBamFiles = library.bamFile.file
-        File indexFiles = library.bamFile.index
+
+        if (defined(starIndex)) {
+            call star_task.Star as star {
+                input:
+                    inputR1 = [qc.qcRead1],
+                    inputR2 = select_all([qc.qcRead2]),
+                    outFileNamePrefix = outputDir + "/star/" + sample.id + "-" + readgroup.lib_id + "-" + readgroup.id + ".",
+                    outSAMattrRGline = [rgLine],
+                    indexFiles = select_first([starIndex]),
+                    dockerImage = dockerImages["star"]
+            }
+        }
+
+        if (defined(hisat2Index)) {
+            call hisat2Task.Hisat2 as hisat2 {
+                 input:
+                indexFiles = select_first([hisat2Index]),
+                inputR1 = readgroup.R1,
+                inputR2 = readgroup.R2,
+                outputBam = outputDir + "/hisat2/" + sample.id + "-" + readgroup.lib_id + "-" + readgroup.id + ".bam",
+                sample = sample.id,
+                library = readgroup.lib_id,
+                readgroup = readgroup.id,
+                platform = platform,
+                dockerImage = dockerImages["hisat2"]
+            }
+        }
+        # Choose whether to use the STAR or HISAT2 bamfiles for downstream analyses,
+        # star is taken over hisat2.
+        File continuationBam = select_first([star.bamFile, hisat2.bamFile])
+    }
+
+    call samtools.Merge as samtoolsMerge {
```

Comment:
Is this step still necessary? markduplicates can take multiple BAM files as input.